dns/ddclient/src/opnsense/scripts/ddclient/lib/account/cloudflare.py

```diff
@@ -0,0 +1,180 @@
+    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+"""
+import syslog
+import requests
+from requests.auth import HTTPBasicAuth
+from . import BaseAccount
+
+
+class Cloudflare(BaseAccount):
+    _priority = 65535
+
+    _services = {
+        'cloudflare': 'api.cloudflare.com'
+    }
+
+    def __init__(self, account: dict):
+        super().__init__(account)
+
+    @staticmethod
+    def known_services():
+        return  Cloudflare._services.keys()
+
+    def match(account):
+        if account.get('service') in Cloudflare._services:
+            return True
+        else:
+            return False
+
+    def execute(self):
+        if super().execute():
+            proto = 'https' if self.settings.get('force_ssl', False) else 'http'
+
+            # IPv4/IPv6
+            recordType = None
+            if str(self.current_address).find(':') > 1:
+                #IPv6
+                recordType = "AAAA"
+            else:
+                #IPv4
+                recordType = "A"
+            if self.is_verbose:
+                syslog.syslog(
+                    syslog.LOG_NOTICE,
+                    "Account %s record type (IPv4/v6): %s" % (self.description, recordType)
+                )
+            
+            # get ZoneID
+            url = "%s://%s/client/v4/zones" % (proto, self._services[self.settings.get('service')])
+            req_opts = {
+                'url': url,
+                'params': {
+                    'name': self.settings.get('zone')
+                },
+                'headers': {
+                    'User-Agent': 'OPNsense-dyndns',
+                    'X-Auth-Email': self.settings.get('username'),
+                    'X-Auth-Key': self.settings.get('password')
+                }
+            }
+            response = requests.get(**req_opts)
+            try:
+                payload = response.json()
+            except requests.exceptions.JSONDecodeError:
+                payload = {}
+            if 'success' not in payload:
+                syslog.syslog(
+                        syslog.LOG_ERR,
+                        "Account %s error parsing JSON response [ZoneID] %s" % (self.description, payload)
+                    )
+                return
+            if not bool(payload['success']):
+                syslog.syslog(
+                    syslog.LOG_ERR,
+                    "Account %s error receiving ZoneID [%s - %s]" % (self.description, payload['errors'][0]['code'], payload['errors'][0]['message'])
```

Comment:
it's safer to serialize `payload.get('errors', {})` here as the entry may not exist (more a precaution)